addons/web/static/src/js/widgets/data_export.js

```diff
@@ -55,13 +56,23 @@ var DataExport = Dialog.extend({
 
         var got_fields = new $.Deferred();
         this.$el.find('#import_compat').change(function() {
-            self.$el.find('#fields_list').empty();
             self.$el.find('#field-tree-structure').remove();
             var import_comp = self.$el.find("#import_compat").val();
             self.rpc("/web/export/get_fields", {
                 model: self.dataset.model,
                 import_compat: !!import_comp,
             }).done(function (records) {
+                if (import_comp) {
+                    var export_list = self.$("#fields_list option").map(function (res) {
+                        console.log(this);
+                        for (var i = 0; i < records.length; i++) {
+                            if (this.value === records[i].id || $(this).attr("import_compatible") === "false") {break;}
+                        }
```

Comment:
why not just use `_.find`? or `_.findWhere` to see if there's a record whose id matches `this.value`?